CoreScriptsRoot/Modules/Settings/Pages/GameSettings.lua

```diff
@@ -103,10 +103,38 @@ local function Initialize()
 
 		settingsDisabledInVR[this.FullscreenEnabler] = true
 
-		local fullScreenSelectionFrame = this.FullscreenEnabler.SliderFrame and this.FullscreenEnabler.SliderFrame or this.FullscreenEnabler.SelectorFrame
-
+		local Debounce = false
+		
 		this.FullscreenEnabler.IndexChanged:connect(function(newIndex)
-			GuiService:ToggleFullscreen()
+			if Debounce then return end
+			Debounce = true
+			if newIndex == 1 then
+				if not GameSettings:InFullScreen() then
+					GuiService:ToggleFullscreen()
+					this.FullscreenEnabler:SetSelectionIndex(1)
+				end
+			elseif newIndex == 2 then
+				if GameSettings:InFullScreen() then
+					GuiService:ToggleFullscreen()
+					this.FullscreenEnabler:SetSelectionIndex(2)
+				end
+			end
+			Debounce = false
+		end)
+
+		GameSettings.FullscreenChanged:connect(function(isFullScreen)
```

Comment:
If I leave the `IndexChanged` function as it was before then it would rapidly toggle the fullscreen on and off. Due to the `GuiService:ToggleFullscreen()` firing the `FullscreenChanged` which then invokes the `IndexChanged` again.